File: src/main/java/com/anur/config/ElectConfigHelper.java
package com.anur.config;

/**
 * Created by Anur IjuoKaruKas on 4/4/2019
 */
public class ElectConfigHelper extends ConfigHelper {

    public static int getElectionTimeoutMs() {
        return getConfig(ConfigEnum.ELECT_ELECTION_TIMEOUT_MS, Integer::valueOf);
    }

    public static int getVotesBackOffMs() {
        return getConfig(ConfigEnum.ELECT_VOTES_BACK_OFF_MS, Integer::valueOf);
    }

    public static int getHeartBeatMs() {
        return getConfig(ConfigEnum.ELECT_HEART_BEAT_MS, Integer::valueOf);
    }
}

File: src/main/java/com/anur/config/ElectConfigHelper.kt
package com.anur.config

import com.anur.config.common.ConfigurationEnum
import com.anur.config.common.ConfigHelper

/**
 * Created by Anur IjuoKaruKas on 2019/7/5
 */
class ElectConfigHelper : ConfigHelper() {

    companion object : ConfigHelper() {
        fun getElectionTimeoutMs(): Int {
            return getConfig(ConfigurationEnum.ELECT_ELECTION_TIMEOUT_MS) { Integer.valueOf(it) } as Int
        }

        fun getVotesBackOffMs(): Int {
            return getConfig(ConfigurationEnum.ELECT_VOTES_BACK_OFF_MS) { Integer.valueOf(it) } as Int
        }

        fun getHeartBeatMs(): Int {
            return getConfig(ConfigurationEnum.ELECT_HEART_BEAT_MS) { Integer.valueOf(it) } as Int
        }
    }
}